# Meeting with Hans on Zoom 
2-2:45pm Wednesday Feb 19th, 2025
## Agenda
 ### Research question: 
 - How does does social factors of people living with or without HIV affect the gut microbiome?
 -  How does gut microbiome of different labor and non-labor workers in Zimbabwe of different sexes in correlative with HIV outcome?



 ### Data Overview (Susana)
 
 ### Experimental aims:
Version 1

 * Aim 1: Evaluate gut microbiome composition of HIV-positive and HIV-negative patients belonging to to different occupational labor types 

 * Aim 2:  Evaluate the sex differences of gut microbiome composition between HIV-positive and HIV-negative patients. 

 * Aim 3: Investigate microbiome differences in the combined sex and labour subgroups to investigate the interactive effect of sex and labour on the gut microbiome in HIV-positive and HIV-negative patients.


Version 2
 
 * Aim 1: Investigate how multiple social factors (labour type, education and HOH status) influence the gut microbiome (GM) in HIV-positive and HIV-negative patients. (using PERMANOVA)

 * Aim 2:  Identify which social factor has the most significant impact on the gut microbiome in HIV-positive and HIV-negative patients (using multivariable regression analyses)

 * Aim 3: Investigate functional changes in the gut microbiome linked to the key social factor(s) identified in Aim 2 using differential abundance analysis (DESeq2).
   

 ### Outline the planned analysis (alpha, beta diversity, etc.)

 Version 1

 * Aim 1: Filter data to subdivide patient entries into HIV-positive and HIV-negative groups.
- Within HIV-positive and negative subgroups, further categorize patients into labor subtype - labor_carrying, labor_specialized, non_labour, unemployed
- Assess the alpha diversity of microbiome composition using Shannonâ€™s index (richness) within these specialized labor groups within HIV positive and negative patient subsets
- Assess the beta diversity of microbiome composition using weighted unifrac (community dissimilarity with phylogenetic relationships) between these specialized labor groups across both HIV positive and negative patient subsets 

Two different analyses:
- Same labor groups but comparing HIV+ vs HIV-Different labor groups within same HIV+ and HIV- cohort.
- Conduct an indicator species analysis to identify characterizing taxa that are indicative of a labor category

* Aim 2: Gut microbiome analysis using Indicator Species Analysis to identify if a particular bacterial taxa is consistently present and abundant in one of our subgroups but not in others
- On female and male HIV positive patients
- On female and male HIV negative patients

* Aim 3:
- Use a Chi-square test (or logistic regression) to assess the association between gender and labor status in HIV patients.
- Run DESeq2 analysis to identify microbial taxa that are differentially abundant between the subgroups

 
 
 ### Feedback from Hans
 - He is having a lot of fun at the beach in LA while we are all suffering in Vancouver
 - Hans is reiterating that there is an alternative grading rubric, but is emphasizing moving forward with the standard rubric
 - The rarefraction curve looks great
 - RE Evelyn's suggestion: Hans says we would have to keep as many of those variables as desired (discard the ones that have less information)
 - We can use logistical regression for categorical statistical analysis, we are not bound by what is taught in this course. We can use whatever we find and think is  appropriate.
   
Aims
- RE V2 Aim 1: Good. 1) How many groups would we have for labour_type, education and HOH_STATUS (2, 3, 2 respsectively). 2) Are we going to analyze them separately or all together (Han thinks separetely is a better approach). Hans really likes V2 Aim 1 more than V1 Aim 1, but it needs to be framed slightly differently. "Investigate how different social factors (not multiple, can omit mentioning the factors) influence gut microbiome diversity in HIV+ and HIV- patients (or HIV outcome)."
- Focus on alpha and beta diversity, PCoA plots and look at how different social factors shape the gut microbiome.
- Qualitative approach with multiple panels (each a different factor) and look at which clustering is the best, and investigate the ones that show interesting patterns
- Quantitative numbers are the "explained variance", qualitative is the clustering (including 95% CI)
- High level analysis (all factors) --> choose interesting factors, if we don't see anything, just mention that none of them do and conitnue with negative data
- RE Tien's suggesting using an Advanced R package: Hans thinks Aim 1 is a low hanging fruit for a/b diversity, we could use that package as Aim 2 instead because currently, Aim 2 is the answer to Aim 1
- RE Negative data: we are finding there are no differences which is just as important as finding differences
- For aim 2, we can instead look at differentially abundant microbes, look at predicated pathways/genes that are differentially representated across the samples and correlate with social factors, core microbiome, indicator species analysis (or split into 2 more!)
- For aim 3 --> functional potential
- Another aim can be literature review (links between HIV and microbiome and discuss) 
- If there is a LOT of negative data, lack of signifiance in aim 1 diversity doesns't mean that we don't find diversity / different abundance in aim 2, and we can also combine social factors
- Tien's microphone is acting up again. RE we were to set up 20 screens to run background analysis: "HOW MANY??" "WE JUST WANNA SCAN ALL THE SOCIAL FACTORS" "I THINK ITS GOING TO BE A BIT TOUGH TO DO, CAN YOU JUST USE A FOR LOOP + ALWAYS SAVE STUFF LOCALLY INCASE THINGS GO VERY WRONG". Recommendation for maximum screens: run things as needed, more screens may not be faster if 20 screens is working at 1/20th the speed.
- Rarefaction depth is based on all samples (we filtered out samples but they don't affect our selected categories), whatever we are doing right now is fine
- 1 MINUTE LEFT ON ZOOM ðŸ˜¥
- NO MORE QUESTIONS WE GOOD ðŸ˜Ž
- NO ZOOM IS GONE
