# LN01 - QIIME 2 Demultiplexing: HIV dataset 

Feb 20th, 2025

## Purpose:
To import and sort the 16S rRNA sequences based on barcode information in QIIME2 

## Material: 
1. QIIME2
2. HIV_metadata.tsv (Path in server: /mnt/datasets/project_2/HIV)

## Method:
1. Connect to Cisco Anyconnect Secure VPN. Open the terminal and login to MICB 475 class server using the provided login credentials.
2. Create a dedicated directory for project 2: /HIV
3. Import and demultiplex the dysautonomia dataset using pair-end to create a qza file with
demultiplexed samples 
5. Create a visualization for the demultiplexed samples
6. Move the demux.qzv file to your local computer directory and view it using the QIIME2

## Code: 
[QIIME2 Code](QIIME2-Analysis/QIIME2-Data-Processing-Script.txt)

## Output files:
1. Path in server: /HIV/demux.qza
2. Path in server: /HIV/demux.qzv - [file](QIIME2-Analysis/Qiime-output-files/demux.qzv) is also uploaded to this repository
   
## Results: 
The same result is generated from samples sequenced from both forward and reverse primers
1. Total number of reads:  
2. Total number of samples: 281
3. Range of sequencing depth: 
4. Maximum read length (bp): 
5. All the reads the same length of  bp

Sequence counts
> <img src="/QIIME2-Analysis/Figures/Demultiplexed sequence counts summary.png" height="200">

Sequence lengths
> <img src="..." height="300">

Attached images are generated from [https://view.qiime2.org/](https://view.qiime2.org/)

## Discussion:
1. The demultiplexed sequence summary visualized by QIIME2 indicated that our samples had read lengths ranging from 217 to 251 nucleotides (nts).
2. Most of our samples had a read length of 251 nts (â‰¥75%) while 9% and 2% were at 243 and 217 nts, respectively. The read quality for both 243 and 251 nts was reasonable (Figure 1). 
## Future direction:
1. Denoise sequences using the selected truncation length of either 243 or 251 bp and determine ASVs with DADA2
