# LN04 - RStudio convert to phyloseq object

Feb 28th, 2025

## Purpose:
To import and convert metadata, phylogenetic tree, otu table, and taxonomy of HIV dataset to phyloseq object. 
To remove samples less than 100 reads, ASVs < 5 counts, any mitochondrial or chloroplast sequences, and remove NAs.
To rarefy the phlyoseq object to match with our general data processing step on QIIME2.

## Material: 
1. RStudio
2. QIIME2 
3. tree.nwk
4. table-hiv.txt
5. hiv_metadata.tsbv
6. taxonomy.tsv

## Method:
1. Connect to Cisco Anyconnect Secure VPN. Open the terminal and login to MICB 475 class server using the provided login credentials.
2. Convert taxonomy, otu table, and phylogenic tree to appropriate format for RStudio use.
3. Move the converted files to your local computer directory and view it using the QIIME2
4. Convert files to phyloseq objects with appropriate filtering
5. remove samples less than 100 reads, ASVs < 5 counts, mitochondrial/choloroplasts sequences, and remove NAs
6. Rarefy phyloseq object to appropriate read length done in QIIME2 (12,082)

## Code: 
[RStudio code](RStudio/Aim1/Phyloseq-conversion.Rmd)
QIIME2 codes were applied from MICB 475 lecture
## Output files:
Files are saved on Team OneDrive server
GitHub:[HIV_final](RStudio/Aim1/hiv_final.RData)
| [HIV_rare](RStudio/Aim1/hiv_rare.RData)
   
## Results: 
* Phyloseq object is converted 

## Discussion:
* Materials are now ready to perform alpha beta diversity analysis for aim 1
